tests/zeroconfig/windows/windows_iis_test.go

```diff
@@ -99,23 +100,40 @@ func requireNoErrorExecCommand(t *testing.T, name string, arg ...string) {
 	require.NoError(t, err)
 }
 
-func requireHTTPGetRequestSuccess(t *testing.T, url string) {
+// Returns status code of 0 when an error is hit. The error contains useful information
+// when the status code is 0.
+func checkHTTPGetRequest(t *testing.T, url string) (int, error) {
 	httpClient := &http.Client{}
 	req, err := http.NewRequest("GET", url, nil)
-	require.NoError(t, err)
+	if err != nil {
+		return 0, err
+	}
 	resp, err := httpClient.Do(req)
-	require.NoError(t, err)
+	if err != nil {
+		return 0, err
+	}
+
 	defer resp.Body.Close()
-	require.Equal(t, http.StatusOK, resp.StatusCode)
+	return resp.StatusCode, err
 }
 
 func testExpectedTracesForHTTPGetRequest(t *testing.T, otlp *testutils.OTLPReceiverSink, url string, expectedTracesFileName string) {
-	requireHTTPGetRequestSuccess(t, url)
```

Comment:
We could just move this under `assert.Eventually(t, func() bool {`. I believe it's going to be the same result. No need for extra code

Reply:
Oh I missed that it's `assert.Eventually`. If we change it to `assert.EventuallyWithT` then this call can be moved as is.

```
		if err != nil {
			t.Log(err)
			return false
		}
```
would be `assert.NoError(t, err)` or `require.NoError(t, err)`

Reply:
I don't think I understood exactly what you're suggesting here, but I've pushed my best guess. Let me know how it differs (if so) from your suggestion.

Reply:
You've got it right. The only difference is that the `requireHTTPGetRequestSuccess` function is removed, but that's alright